### ISSUE TITLE
bsu.sh gives out-of-memory errors

### ISSUE THREAD

**Issue by ltutar** [biemond/biemond-orawls#323]
The new bsu patch p21984589_1036_Generic.zip can not be installed because of out-of-memory error.
```
Notice: /Stage[main]/Profiles::Wls/Orawls::Bsu[S8C2]/File[/tmp/install/p21984589_1036_Generic.zip]/ensure: defined content as '{md5}b218f4caa0261d29537cc723de475162'
Notice: /Stage[main]/Profiles::Wls/Orawls::Bsu[S8C2]/Exec[extract p21984589_1036_Generic.zip]/returns: executed successfully
Exception in thread "main" java.lang.OutOfMemoryError: Java heap space
	at java.util.LinkedHashMap.newValueIterator(LinkedHashMap.java:418)
	at java.util.HashMap$Values.iterator(HashMap.java:1038)
	at java.util.AbstractCollection.toArray(AbstractCollection.java:137)
	at java.util.ArrayList.addAll(ArrayList.java:559)
	at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.getProperties(SchemaTypeImpl.java:705)
	at com.bea.cie.common.dao.xbean.XBeanDataHandler.loadPropertyMap(XBeanDataHandler.java:775)
	at com.bea.cie.common.dao.xbean.XBeanDataHandler.<init>(XBeanDataHandler.java:99)
	at com.bea.cie.common.dao.xbean.XBeanDataHandler.createDataHandler(XBeanDataHandler.java:559)
	at com.bea.cie.common.dao.xbean.XBeanDataHandler.getComplexValue(XBeanDataHandler.java:455)
	at com.bea.plateng.patch.dao.cat.PatchCatalogHelper.getPatchDependencies(PatchCatalogHelper.java:442)
	at com.bea.plateng.patch.dao.cat.PatchCatalogHelper.getPatchDependencies(PatchCatalogHelper.java:464)
	at com.bea.plateng.patch.dao.cat.PatchCatalog.getPatchDependencies(PatchCatalog.java:56)
	at com.bea.plateng.patch.dao.cat.PatchCatalogHelper.getInvalidatedPatchMap(PatchCatalogHelper.java:1621)
	at com.bea.plateng.patch.PatchSystem.updatePatchCatalog(PatchSystem.java:436)
	at com.bea.plateng.patch.PatchSystem.refresh(PatchSystem.java:130)
	at com.bea.plateng.patch.PatchSystem.setCacheDir(PatchSystem.java:201)
	at com.bea.plateng.patch.Patch.main(Patch.java:306)
Info: bsu_patch result: Checking for conflicts...
No conflict(s) detected

Starting installation of Patch ID: S8C2

```
When I modify bsu.sh to
```
MEM_ARGS="-Xms256m -Xmx1024m"

```
and then add the yaml lines
```
bsu_instances:
  'S8C2':
    patch_id:                "S8C2"
    patch_file:              "p21984589_1036_Generic.zip"
    log_output:              true
    remote_file:             true
```
it works.
```
Notice: /Stage[main]/Profiles::Wls/Orawls::Bsu[S8C2]/Exec[extract p21984589_1036_Generic.zip]/returns: executed successfully
Info: bsu_patch result: Checking for conflicts...
No conflict(s) detected

Starting installation of Patch ID: S8C2
Installing /opt/wls/middleware11g/utils/bsu/cache_dir/S8C2.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/BUG21984589_1036013.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/com.bea.core.apache.commons.fileupload_1.0.0.0_1-3-1.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/com.bea.core.stax2_2.0.0.0_3-0-3.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/glassfish.jaxb.xjc_1.2.0.0_2-1-14.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/glassfish.jaxb_1.2.0.0_2-1-14.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/glassfish.jaxp_1.4.5.0.jar
Extracting /opt/wls/middleware11g/patch_wls1036/patch_jars/glassfish.jaxws.mimepull_1.1.0.0_1-3-8.jar
Updating /opt/wls/middleware11g/patch_wls1036/profiles/default/sys_manifest_classpath/weblogic_patch.jar
Old manifest value: Class-Path=
New manifest value: Class-Path=../../../patch_jars/BUG21984589_1036013.jar ../../../patch_jars/com.bea.core.apache.commons.fileupload_1.0.0.0_1-3-1.jar ../../../patch_jars/com.bea.core.stax2_2.0.0.0_3-0-3.jar ../../../patch_jars/glassfish.jaxb.xjc_1.2.0.0_2-1-14.jar ../../../patch_jars/glassfish.jaxb_1.2.0.0_2-1-14.jar ../../../patch_jars/glassfish.jaxp_1.4.5.0.jar ../../../patch_jars/glassfish.jaxws.mimepull_1.1.0.0_1-3-8.jar
Backing up /opt/wls/middleware11g/wlserver_10.3/common/deployable-libraries/jsf-2.0.war into /opt/wls/middleware11g/patch_wls1036/backup/backup.jar

```

Is it possible to modify the memory settings of bsu through this module?

**Comment by biemond** [comment]
Yeah, 

it is fixed in this commit 1b12d8222928136c76a4b0b7dd6b2e3bcadef9b3
For now only available on github. 

I will push soon push a new release

**Comment by ltutar** [comment]
It works. Thank you very much
```
Result: Success


Info: bsu_patch done
```
